core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -44,7 +45,9 @@ object RequestChannel extends Logging {
     byteBuffer
   }
 
-  case class Request(processor: Int, connectionId: String, private var buffer: ByteBuffer, startTimeMs: Long, securityProtocol: SecurityProtocol) {
+  case class Session(principal: Principal, host: String)
+
+  case class Request(processor: Int, connectionId: String, session: Session = null, private var buffer: ByteBuffer, startTimeMs: Long, securityProtocol: SecurityProtocol) {
```

Comment:
Normally, one would use `Option[Session]` instead of a nullable one unless efficiency is of particular importance in that use-case. Is this one of those cases?